python/ray/tune/trainable.py

```diff
@@ -178,7 +178,7 @@ def __init__(self, config=None, logger_creator=None):
                         "slow to initialize, consider setting "
                         "reuse_actors=True to reduce actor creation "
                         "overheads.".format(setup_time))
-        self._local_ip = ray.services.get_node_ip_address()
+        self._local_ip = self.get_current_ip()
```

Comment:
This doesn't solve the issue where network disconnect crashes the trial, does it? From what I see, `get_current_ip` just calls the same `ray.services.get_node_ip_address`?

Reply:
No; it doesn't look like #7686 is easy to resolve (requires some changes to ray core). 

This patch is for people who have 2 different IPs and want to override the IP address retrieval.

Reply:
Awesome, I think this should be ready to merged then 🙂